### PR TITLE
JSCS: Require spaces around operators

### PR DESCRIPTION
**Pull request by knownasilya**
Except for before a `,`.

_Note: The build will probably break, because there are violations of the new rules._

``` js
// bad
var a = b+c;
a = b+ b;
if (test> 0) {
}
function (abc ,def) {
}

// good
var a = b + c;
a = b + b;
if (test > 0) {
}
function (abc, def) {
}
```

### CODE REVIEW

**Comment by knownasilya** [comment]
@rwjblue @stefanpenner if this will be merged, I can go through and fix the code to conform to these new rules.

**Comment by rwjblue** [comment]
@knownasilya - Can you show examples of good and bad versions of this?  Also update the styleguide with a little description?  Once that is done, ping me back and I'll review and let you know :+1: / :-1: before you update the code all over.

**Comment by knownasilya** [comment]
@rwjblue is that good enough?

**Comment by rwjblue** [comment]
Yes, but the `function` examples are confusing to me.  This doesn't seem to affect parens, right?

**Comment by knownasilya** [comment]
@rwjblue nope, just the commas. Maybe I should add a "disallow a space" before a comma.

**Comment by tomdale** [comment]
This doesn't affect `this.methodWithCallback(function() { … });` does it?

**Comment by tomdale** [comment]
I am strongly opposed to requiring a space between `function` and `()`.

**Comment by rwjblue** [comment]
@tomdale - Yes, completely agreed, which is what I was clarifying.

**Comment by stefanpenner** [comment]
> I am strongly opposed to requiring a space between function and ().

likewise, is it possible for us to explicitly enforce this with JSCS?

**Comment by rwjblue** [comment]
> likewise, is it possible for us to explicitly enforce this with JSCS?

Yes, it is already enforced (by https://github.com/emberjs/ember.js/commit/5ce22690bf02129378f61a5fd0ff079bfd595de2).

**Comment by stefanpenner** [comment]
> Yes, it is already enforced (by 5ce2269).

:)

File: STYLEGUIDE.md [hunk 1]
```diff
@@ -113,6 +113,21 @@ function foo() {
...
+```javascript
+// assignments
+var foo = bar + 'a';
+
+// conditionals
+if (foo === 'bara') {
+}
+
+// parameters
+function (test, foo) {
```
Comment (tomdale) [inline STYLEGUIDE.md]: I guess we should update this example to follow the styleguide then. ;)

Comment (rwjblue) [inline STYLEGUIDE.md]: Hehe, agreed.

**Comment by knownasilya** [comment]
In my opinion, `function () {}` is more consistent with `function named() {}`, since if you remove the `named`, you still have a space. But what is my opinion :wink:

**Comment by knownasilya** [comment]
@rwjblue ping

**Comment by jayphelps** [comment]
Boy I would love this to be merged....I'll gladly fix the ones that fail, if @knownasilya isn't available. This and lack of statement spaces, `if()`, bug the shizit out of me cause I'm anal-retentive...Now if only we could convince you all to see the light of `function ()`... some day.

Also...this comment is basically just serving as another ping :yum:

**Comment by knownasilya** [comment]
@rwjblue is there anything blocking this? I can work on the PR to fix the cases, and we can merge one  after the other if this is the issue.

**Comment by tomdale** [comment]
This looks good to me. :+1:

**Comment by stefanpenner** [comment]
@knownasilya we have a +1 form señor @tomdale can we get a rebase + passing test?

**Comment by knownasilya** [comment]
@stefanpenner will get on it today, hopefully!

**Comment by knownasilya** [comment]
@stefanpenner I think that's it.

**Comment by knownasilya** [comment]
Ok, rebased.

**Pull request merged**